pandas/tests/test_optional_dependency.py

```diff
@@ -50,3 +52,11 @@ def test_no_version_raises():
 
     with pytest.raises(ImportError, match="Can't determine .* fakemodule"):
         import_optional_dependency(name)
+
+
+@td.skip_if_installed("tables")
```

Comment:
in the interest of better coverage, would it not be better to monkeypatch `builtins.__import__` as was done in #26685 for tests like this rather than skip?